sonoff.server.js

```diff
@@ -55,7 +55,7 @@ server.get('/devices/:deviceId/status', function (req, res) {
 });
 
 //switch the device
-server.get('/devices/:deviceId/:state', function (req, res) {
+server.get('/devices/:deviceId/:index/:state', function (req, res) {
```

Comment:
Sorry, but this would break the compatibility with the existing implementation. Could you create a secound "get" with this parameter? You might want to export the functionality to a new function and call it by both paths.

function devices(deviceId, index, state, res) {
 ....
}

server.get('/devices/:deviceId/:index/:state', function (req, res) {
  devices(req.params.deviceId, req.params.index, req.params.state, res)
}

server.get('/devices/:deviceId/:state', function (req, res) {
  devices(req.params.deviceId, req.params.index, req.params.state, res)
}